### PR TITLE
fix(material-experimental/mdc-form-field): fix outline notch width

### PR DESCRIPTION
**Pull request by devversion**
This commit fixes a couple of issues that have been introduced
by accident over time:

1. The outline notch width is incorrectly calculated since https://github.com/angular/components/pull/21676. This happens because we manually override
  the label's typography level to `body1` but didn't account for the `font-size` set
  by the `notched-outline`. This resulted in the computed label width being incorrect.

2. https://github.com/angular/components/pull/22089 added `line-height:
   0` to fix textarea's accidentally expanding the infix vertically.
   This breaks custom controls and their alignment. Textarea's seem to behave a
   little different compared to a native input, and the actual fix seems
   to be to fix the alignment from `top` to `middle` so that the infix
   does not try to add additional padding (due to the inherited `body1`
   line-height and to satisfy `min-height`). Example:
   https://jsfiddle.net/rpvm4bkL/6/.

3. Typography has been moved with
   https://github.com/angular/components/pull/21676 to the
   input control (notice how `font: inherit` is removed too). This
   breaks custom form controls that rely on the typography provided
   by the form-field (these controls would just inherit typography)

4. Not an actual fix, but a cleanup since 21ab17fb1eea3a7e91463bcacc282ef455f7845b
  added a class to avoid styles from the form-field leaking into
  standalone inputs. The class has been updated to be more specific that
  it only matches form-control inputs, and not _any_ type of form
  control.

Also cleans up logic for the label offset computation. Instead of hard-coding values extracted from MDC, we should have a constant describing where this value comes from. Lastly, deletes a `.swp` file that has sneaked into the repo.

### CODE REVIEW

File: src/material-experimental/mdc-form-field/_mdc-text-field-textarea-overrides.scss [hunk 1]
```diff
@@ -10,7 +10,11 @@
 @mixin private-text-field-textarea-overrides() {
   // Ensures that textarea elements inside of the form-field have proper vertical spacing
   // to account for the floating label. Also ensures that there is no vertical text overflow.
-  .mat-mdc-textarea-input {
+  .mat-mdc-form-field-input-control.mat-mdc-textarea-input {
+    // Set the vertical alignment for textareas inside form fields to be the middle. This
+    // ensures that textareas do not stretch the infix container vertically without having
+    // multiple rows of text. See: https://github.com/angular/components/pull/22089.
+    vertical-align: middle;
```
Comment (mmalerba) [inline src/material-experimental/mdc-form-field/_mdc-text-field-textarea-overrides.scss:17]: Are you sure this doesn't mess up prefix/suffix icon alignment?

Comment (devversion) [reply src/material-experimental/mdc-form-field/_mdc-text-field-textarea-overrides.scss:17]: I've tested with a text suffix/prefix and icons, looks good. Here is a screenshot for you to verify:

Comment (crisbeto) [reply src/material-experimental/mdc-form-field/_mdc-text-field-textarea-overrides.scss:17]: Is it just me or are there some extra lines in that screenshot? 
122447133-b815fe00-cfa3-11eb-8865-0c7dd87bd938

Comment (devversion) [reply src/material-experimental/mdc-form-field/_mdc-text-field-textarea-overrides.scss:17]: This is unrelated to this PR (can be observed in Miles' form-field demo too; https://mmalerba-demo-1.web.app/mdc-input). 

I'll have a look at this too; it's also observable on the MDC textfield itself but not as obvious.

**Review by crisbeto — APPROVED** [review]
LGTM, but leaving to Miles for final approval.

File: src/material-experimental/mdc-form-field/_mdc-text-field-textarea-overrides.scss (hunk 1 above)
Comment (mmalerba) [reply src/material-experimental/mdc-form-field/_mdc-text-field-textarea-overrides.scss:17]: Yeah I've noticed that extra line before too, it comes from the MDC styles. I remember having a similar issue back when I was writing our outline form field too.

**Review by mmalerba — APPROVED** [review]
LGTM

**Comment by mmalerba** [comment]
@devversion needs rebase

**Comment by devversion** [comment]
@mmalerba Done.

**Comment by wagnermaciel** [comment]
I believe the additional padding is causing the text area to overflow

#### Before
Screen Shot 2021-07-13 at 11 06 16 AM

#### After
Screen Shot 2021-07-13 at 11 06 23 AM

This is causing a lot of screenshot test failures internally

**Comment by wagnermaciel** [comment]
Did some more digging. Turns out there used to be `resize: none` on the textareas coming from `cdk/text-field/_index.scss`. They are now getting `resize: auto` from `material-experimental/mdc-form-field/_mdc-text-field-textarea-overrides.scss`. That plus some of the other styles from there are probably what's breaking the screenshot tests

**Comment by devversion** [comment]
@wagnermaciel Nice findings. It makes sense that this happens now because our MDC styles for the textarea have increased in specificity. I'd need to look how we can make this work. It seems like even with same specificity of the CDK textarea styles this is prone to break.

**Comment by wagnermaciel** [comment]
Does that mean this an expected failure? Is this how the text input is meant to look now?

**Comment by devversion** [comment]
@wagnermaciel It's not expected. I didn't realize this was happening because our examples of the form-field are quite large. Updated the PR to let `cdk-textarea-autosize` override the `resize` CSS property to `none`. Should be good for another presubmit.

File: src/material-experimental/mdc-form-field/_form-field-theme.scss [hunk 2]
```diff
@@ -72,14 +72,30 @@
...
-    // MDC uses the `subtitle1` level for the input label and value, but the spec shows `body1` as
-    // the correct level.
-    .mat-mdc-form-field-control,
-    .mat-mdc-form-field label,
-    .mat-mdc-form-field-text-prefix,
-    .mat-mdc-form-field-text-suffix {
+    // MDC uses `subtitle1` for the input value, placeholder and floating label. The spec
+    // shows `body1` for text fields though, so we manually override the typography.
+    // Note: Form controls inherit the typography from the parent form field.
+    .mat-mdc-form-field,
```
Comment (mmalerba) [inline src/material-experimental/mdc-form-field/_form-field-theme.scss:78]: I think changing these selectors broke some of the typography. I'm seeing a few screenshots in g3 where the font changed. Can we change them back?

**Pull request merged**